internal/providers/credentials/github_app_jwt.go

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package credentials
+
+import (
+	"crypto/rsa"
+	"fmt"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+// CreateGitHubAppJWT creates a JWT token for a GitHub App
+func CreateGitHubAppJWT(appId int64, privateKey *rsa.PrivateKey) (string, error) {
+	// Create the Claims
+	claims := jwt.MapClaims{
+		"iat": time.Now().Unix(),
+		"exp": time.Now().Add(time.Minute * 10).Unix(),
+		"iss": appId,
```

Comment:
isn't the `iss` supposed to be a string? even checking https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app suggests so.

Reply:
Both work, but I made it a string to be more correct